### ISSUE TITLE
Getting Error for eth get transaction receipt 

### ISSUE THREAD

**Issue by hbisht-airstack** [NethermindEth/nethermind#5805]
While doing a curl on the server for eth get transaction receipt 
`curl --data '{"method":"eth_getTransactionReceipt","params":[["0x80757153e93d1b475e203406727b62a501187f63e23b8fa999279e219ee3be71"]],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545`

getting error as of below.
`{"jsonrpc":"2.0","error":{"code":-32603,"message":"Internal error","data":"System.NullReferenceException: Object reference not set to an instance of an object.\n   at Nethermind.Db.DbExtensions.Get(IDb db, Keccak key)\n   at Nethermind.Blockchain.Receipts.PersistentReceiptStorage.FindBlockHash(Keccak txHash)\n   at Nethermind.Blockchain.Receipts.FullInfoReceiptFinder.FindBlockHash(Keccak txHash)\n   at Nethermind.Facade.BlockchainBridge.GetReceiptAndEffectiveGasPrice(Keccak txHash)\n   at Nethermind.JsonRpc.Modules.Eth.EthRpcModule.eth_getTransactionReceipt(Keccak txHash)\n   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)\n   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)"},"id":1}`

**Comment by MarekM25** [comment]
I guess we can't find receipts but it would be nice to return better error message here.
cc: @emlautarom1

**Comment by emlautarom1** [comment]
Actually, this problem is not due to missing receipts but an error on the parameters of the call:

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "eth_getTransactionReceipt",
  "params": [["0x80757153e93d1b475e203406727b62a501187f63e23b8fa999279e219ee3be71"]]
}
```

Here, you are sending a single parameter that is an array, which contains a single string "0x8075...". You can see on the [official spec](https://ethereum.github.io/execution-apis/api-documentation/) that this method expects a single parameter, which should be a 32 byte hex value.

The solution is to make the request with the following body:

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "eth_getTransactionReceipt",
  "params": ["0x80757153e93d1b475e203406727b62a501187f63e23b8fa999279e219ee3be71"]
}
```

Note that `params` is now an array with a single string element.

**Comment by emlautarom1** [comment]
Nevertheless, the error you're getting is due to incorrect parameter handling on Nethermind. Instead, you should be getting the following error response:

```json
{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params"},"id":1}
```

A fix is implemented in #6104.
